modules/tinymce/src/core/main/ts/content/GetContentImpl.ts

```diff
@@ -22,9 +22,15 @@ const getContentFromBody = (editor: Editor, args: GetContentArgs, body: HTMLElem
   if (args.format === 'raw') {
     content = Tools.trim(TrimHtml.trimExternal(editor.serializer, body.innerHTML));
   } else if (args.format === 'text') {
-    // return empty string for text format when editor is empty to avoid bogus elements being returned in content
-    content = Zwsp.trim(body.innerText);
-    content = content === '\n' ? '' : content;
+    content = getContentFromBody(editor, { ...args, format: 'html' }, body);
+
+    if (content === '' || !Type.isString(content)) {
+      return content;
+    }
+
+    const parseBody = document.createElement('body');
+    parseBody.innerHTML = trimEmptyContents(editor, content);
+    content = Zwsp.trim(parseBody.innerText.replace(/\xa0/g, ''));
```

Comment:
This isn't going to work because it will remove all `&nbsp;` characters from the output. For example try the following HTML:
```
<p>this is test&nbsp; content</p>
```

Instead of having 2 spaces between `test` and `content` it will only have 1 space. Also as a side note, we normally use `\u00a0` instead of `\xa0` for nbsp characters.

Reply:
@HAFRMO have you considered or tried replacing the bogus elements with a nonbreaking zero-width space to solve this instead? As the problem if I understand correctly was that with `<br>` that added a newline and the problem with `&nbsp;` was that it added additional spaces. However if we use `\uFEFF` that's already an internal character and we can then strip it safely from the text after getting the `innerText`.

I tested this theory via the HTML in https://fiddle.tiny.cloud/5eiaab and it appeared to work fine on Chrome.